### PR TITLE
Rebindable support for AAs.

### PR DESCRIPTION
**Pull request by japplegame**

### CODE REVIEW

**Comment by DmitryOlshansky** [comment]
LGTM

File: std/typecons.d [hunk 1]
```diff
@@ -1562,7 +1562,7 @@ risks usually associated with $(D cast).
 Params:
     T = An object, interface, or array slice type.
  */
-template Rebindable(T) if (is(T == class) || is(T == interface) || isDynamicArray!T)
+template Rebindable(T) if (is(T == class) || is(T == interface) || isDynamicArray!T || isAssociativeArray!T)
```
Comment (quickfur) [inline std/typecons.d]: Should we wrap this very long line to something more readable?

**Comment by quickfur** [comment]
Other than that, LGTM.

File: std/typecons.d (hunk 1 above)
Comment (schveiguy) [inline std/typecons.d]: I think it's OK, docs look ok.

**Comment by schveiguy** [comment]
In [docs](http://dtest.thecybershadow.net/artifact/website-eb9635f66087761793d863b836e3677b93b20289-32a7505a808595cb9b937e4ac23d5ac9/web/phobos-prerelease/std_typecons.html#.Rebindable), it says: 

```
However, Rebindable!(T) does not compile if T is a non-class type.
```

This needs updating. also, an example of AA usage may be good.

**Comment by japplegame** [comment]
Should I squash commits?

**Comment by japplegame** [comment]
Docs and formatting fixed.

**Comment by schveiguy** [comment]
Auto-merge toggled on

**Pull request merged**